assets/src/edit-story/components/header/buttons.js

```diff
@@ -74,17 +75,19 @@ function Publish() {
   const {
     state: {
       meta: { isSaving },
-      story: { date },
+      story: { date, storyId },
     },
     actions: { updateStory },
   } = useStory();
 
+  const refreshPostEditURL = useRefreshPostEditURL(storyId);
   const hasFutureDate = Date.now() < Date.parse(date);
 
-  const handlePublish = useCallback(
-    () => updateStory({ properties: { status: 'publish' } }),
-    [updateStory]
-  );
+  const handlePublish = useCallback(() => {
+    updateStory({ properties: { status: 'publish' } });
+    // Note that story ID exists for new posts as well, even if not saved.
+    refreshPostEditURL(storyId);
```

Comment:
Why are you passing the `storyId` in here? The story id doesnt change here?

Reply:
Ah, good point, no need to pass it at all, will change 👍

Reply:
Or actually, now I remember that it was passed since the story context was not available in the util.

Reply:
Why pass it into the hook if you are not going to use it?

Reply:
Okay, now I see what you mean, I got confused with `useRefreshPostEditURL`. 👍

Reply:
0c5f1ee